client/messaging/message-list.tsx

```diff
@@ -49,11 +60,25 @@ const handleUnknown = (msg: Message) => {
 const PureMessageList = React.memo<PureMessageListProps>(({ messages, renderMessage }) => {
   return (
     <Messages>
-      {messages.map(m => {
+      {messages.map((m, index, array) => {
         // NOTE(2Pac): We only handle common messages here, e.g. text message. All other types of
         // messages are handled by calling the `renderMessage` function which should be supplied by
         // each service if they have any special messages to handle.
-        return renderCommonMessage(m) ?? (renderMessage ?? handleUnknown)(m)
+        const messageLayout = renderCommonMessage(m) ?? (renderMessage ?? handleUnknown)(m)
+
+        const nextMessage = array.get(index + 1)
```

Comment:
It's a small thing, but generally I think you should prefer looking *backwards* rather than *forwards* to do things like this. This message only makes sense to show if the "real" message after it is rendered, and if we're looking backwards, that message is *this* one.